### PR TITLE
fix:allow users to identify search is enabled in country dropdown

### PR DESCRIPTION
**Pull request by saiprabhu-dandanayak**
Fixes #`Issue Number`  
_or_  
Fixes `Issue URL`
> [!WARNING]  
> _If no issue exists, please create an issue first, and check with the maintainers if the issue is valid._

## Automation

/ok-to-test tags=""

### :mag: Cypress test results

> [!CAUTION]  
> If you modify the content in this section, you are likely to disrupt the CI result for your PR.

## Communication
Should the DevRel and Marketing teams inform users about this change?
- [ ] Yes
- [ ] No

### Bug : [Country dropdown bug](https://github.com/appsmithorg/appsmith/issues/34074)

I have raised this PR `In-order to allow user to know that , they can search for countries in the country dropdown in CurrencyInputWidget and PhoneInputWidget`

#### a small cursor is blinking before the selected value
image

## Summary by CodeRabbit

- **New Features**
  - Enhanced dropdown control with improved state management for focus and blur events.
  - Added new event handlers to better manage property updates and styling.

### CODE REVIEW

File: app/client/src/components/propertyControls/DropDownControl.tsx [hunk 1]
```diff
@@ -58,6 +61,24 @@ class DropDownControl extends BaseControl<DropDownControlProps> {
     }
   };
 
+  handleOnClick = () => {
+    this.updateProperty(this.props.propertyName, "");
+  };
+
+  handleFocus = () => {
+    this.setState({ previousValue: this.props.propertyValue });
+    if (this.containerRef && this.containerRef.current) {
```
[the remark this answers is not in the document]
Comment (saiprabhu-dandanayak) [reply app/client/src/components/propertyControls/DropDownControl.tsx]: resolved

**Review by marks0351 — CHANGES_REQUESTED** [review]
@saiprabhu-dandanayak can you mention which issue u r working on for this pr?

**Comment by saiprabhu-dandanayak** [comment]
> @saiprabhu-dandanayak can you mention which issue u r working on for this pr?

@marks0351 , https://github.com/appsmithorg/appsmith/issues/34074 this is the issue , i also mentioned it in PR description .

**Comment by marks0351** [comment]
@saiprabhu-dandanayak thanks.
I would suggest you to pick up an issue from our list of [good first issues](https://github.com/appsmithorg/appsmith/issues?q=is%3Aopen+is%3Aissue+label%3A%22Good+First+Issue%22). Also pls do comment on the issue that you want to contribute, to assign the issue to you so that someone in our team is aware of it and can help you run our internal processes to validate a contribution.
For more info pls do check our [contribution guide](https://github.com/appsmithorg/appsmith/blob/release/contributions/CodeContributionsGuidelines.md).

cc: @somangshu

**Comment by saiprabhu-dandanayak** [comment]
> @saiprabhu-dandanayak thanks. I would suggest you to pick up an issue from our list of [good first issues](https://github.com/appsmithorg/appsmith/issues?q=is%3Aopen+is%3Aissue+label%3A%22Good+First+Issue%22). Also pls do comment on the issue that you want to contribute, to assign the issue to you so that someone in our team is aware of it and can help you run our internal processes to validate a contribution. For more info pls do check our [contribution guide](https://github.com/appsmithorg/appsmith/blob/release/contributions/CodeContributionsGuidelines.md).
> 
> cc: @somangshu

@marks0351 , can i know the above mentioned pr is not relavant for the isssue i mentioned is there any other way to do that .

**Pull request closed without merging**